src/index.ts

```diff
@@ -100,6 +82,29 @@ export default class Kumulos {
         }
     }
 
+    private initializePushFeature() {
+        trackOpenFromQuery(this.context);
+        registerServiceWorker(this.context.serviceWorkerPath);
+
+        new PromptManager(
```

Comment:
Is there any risk of GC of the manager now we don't assign it to a property?

Reply:
Shouldnt be a problem as the register is done on navigator.serviceWorker (navigator references the service worker). The registration is accessed by navigator.serviceWorker then. Dont see a reason to keep a reference to the registration.

Reply:
Sorry not reference to the registration, but the PromptManager itself is responsible for drawing soft prompts for web opt in, if it is GC maybe it won't run logic to show the UI elements or handle button presses? (it's possible there's some internal ref or cycle, but just cautious here).

Have we regression tested prompts with this build?

Reply:
Oh I see. Regarding the PromptManager, there were no references to the object from outside. The object shouldn't be destroyed unless there is no "potential" code anymore. So as long as we are awaiting something inside / create another object and pass callbacks (onPromptAccepted={this.onPromptAccepted} for example) the object will stay. It should be cleared when its done execution and no one references it. In general, I don't think that its a good idea when the constructor is the only exposed method but lets refactor this later.